Compilation failure with CGAL <5.5 when compiling with clang

I am not sure there is much we can do about this one. But compiling deal.II with clang against CGAL versions prior to 5.5 fails:
```
/usr/include/CGAL/Intersections_3/internal/Tetrahedron_3_Bounded_3_do_intersect.h:55:23: error: call to 'do_intersect' is ambiguous
    const Boolean b = do_intersect(tr,
                      ^~~~~~~~~~~~
/usr/include/CGAL/Intersections_3/internal/Tetrahedron_3_Triangle_3_do_intersect.h:36:10: note: in instantiation of function template specialization 'CGAL::Intersections::internal::do_intersect_tetrahedron_bounded<CGAL::Simple_cartesian<CGAL::Mpzf>, CGAL::Triangle_3<CGAL::Simple_cartesian<CGAL::Mpzf>>>' requested here
  return do_intersect_tetrahedron_bounded(tr, tet, tr[0], k);
         ^
/usr/include/CGAL/Kernel/function_objects.h:3031:39: note: in instantiation of function template specialization 'CGAL::Intersections::internal::do_intersect<CGAL::Simple_cartesian<CGAL::Mpzf>>' requested here
    { return Intersections::internal::do_intersect(t1, t2, K()); }
                                      ^
/usr/include/CGAL/Filtered_predicate.h:111:12: note: in instantiation of function template specialization 'CGAL::CommonKernelFunctors::Do_intersect_3<CGAL::Simple_cartesian<CGAL::Mpzf>>::operator()<CGAL::Tetrahedron_3<CGAL::Simple_cartesian<CGAL::Mpzf>>, CGAL::Triangle_3<CGAL::Simple_cartesian<CGAL::Mpzf>>>' requested here
    return ep(c2e(args)...);
           ^
/usr/include/CGAL/Static_filtered_predicate.h:52:12: note: in instantiation of function template specialization 'CGAL::Filtered_predicate<CGAL::CommonKernelFunctors::Do_intersect_3<CGAL::Simple_cartesian<CGAL::Mpzf>>, CGAL::CommonKernelFunctors::Do_intersect_3<CGAL::Simple_cartesian<CGAL::Interval_nt<false>>>, CGAL::Cartesian_converter<CGAL::Type_equality_wrapper<CGAL::Cartesian_base_no_ref_count<double, CGAL::Epick>, CGAL::Epick>, CGAL::Simple_cartesian<CGAL::Mpzf>, CGAL::NT_converter<double, CGAL::Mpzf>>, CGAL::Cartesian_converter<CGAL::Type_equality_wrapper<CGAL::Cartesian_base_no_ref_count<double, CGAL::Epick>, CGAL::Epick>, CGAL::Simple_cartesian<CGAL::Interval_nt<false>>, CGAL::NT_converter<double, CGAL::Interval_nt<false>>>, true>::operator()<CGAL::Tetrahedron_3<CGAL::Epick>, CGAL::Triangle_3<CGAL::Epick>>' requested here
    return epicp(aa1.first, aa2.first);
           ^
/usr/include/CGAL/Intersections_3/Tetrahedron_3_Triangle_3.h:32:1: note: in instantiation of function template specialization 'CGAL::Static_filtered_predicate<CGAL::Simple_cartesian<CGAL::Interval_nt<false>>, CGAL::Filtered_predicate<CGAL::CommonKernelFunctors::Do_intersect_3<CGAL::Simple_cartesian<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on>>>, CGAL::CommonKernelFunctors::Do_intersect_3<CGAL::Simple_cartesian<CGAL::Interval_nt<false>>>, CGAL::Exact_converter<CGAL::Epeck, CGAL::Simple_cartesian<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on>>>, CGAL::Approx_converter<CGAL::Epeck, CGAL::Simple_cartesian<CGAL::Interval_nt<false>>>, true>, CGAL::internal::Static_filters_predicates::Do_intersect_3<CGAL::Filtered_kernel_base<CGAL::Type_equality_wrapper<CGAL::Cartesian_base_no_ref_count<double, CGAL::Epick>, CGAL::Epick>>, CGAL::internal::Static_filters<CGAL::Filtered_kernel_base<CGAL::Type_equality_wrapper<CGAL::Cartesian_base_no_ref_count<double, CGAL::Epick>, CGAL::Epick>>>>>::operator()<CGAL::Tetrahedron_3<CGAL::Epeck>, CGAL::Triangle_3<CGAL::Epeck>>' requested here
CGAL_DO_INTERSECT_FUNCTION(Tetrahedron_3, Triangle_3, 3)
^
/usr/include/CGAL/Intersection_traits.h:66:25: note: expanded from macro 'CGAL_DO_INTERSECT_FUNCTION'
    return BOOST_PP_CAT(K().do_intersect_, BOOST_PP_CAT(DIM, _object()(a, b))); \
                        ^
/home/tamiko/workspace/dealii/source/cgal/intersections.cc:558:23: note: in instantiation of function template specialization 'CGAL::do_intersect<CGAL::Epeck>' requested here
            if (CGAL::do_intersect(tet, tria_quad.triangle(f)))
                      ^
/usr/include/CGAL/Intersections_3/internal/Triangle_3_Triangle_3_do_intersect.h:180:1: note: candidate function [with K = CGAL::Simple_cartesian<CGAL::Mpzf>]
do_intersect(const typename K::Triangle_3& t1,
^
/usr/include/CGAL/Boolean_set_operations_2/do_intersect.h:239:13: note: candidate function [with InputIterator = CGAL::Triangle_3<CGAL::Simple_cartesian<CGAL::Mpzf>>, Traits = const CGAL::Simple_cartesian<CGAL::Mpzf>]
inline bool do_intersect(InputIterator begin, InputIterator end, Traits& traits,
            ^
```

This is fixed in the newest version (5.5)

--- a/source/cgal/intersections.cc
+++ b/source/cgal/intersections.cc
@@ -269,7 +269,7 @@ namespace CGALWrappers
     compute_intersection(const std::array<Point<3>, 2> &first_simplex,
                          const std::array<Point<3>, 4> &second_simplex)
     {
-#  if DEAL_II_CGAL_VERSION_GTE(5, 1, 5)
+#  if DEAL_II_CGAL_VERSION_GTE(5, 5, 0)
       std::array<CGALPoint3, 4> pts0;
       std::array<CGALPoint3, 2> pts1;
       std::transform(
@@ -311,7 +311,7 @@ namespace CGALWrappers
     compute_intersection(const std::array<Point<3>, 3> &first_simplex,
                          const std::array<Point<3>, 4> &second_simplex)
     {
-#  if DEAL_II_CGAL_VERSION_GTE(5, 1, 5)
+#  if DEAL_II_CGAL_VERSION_GTE(5, 5, 0)
       std::array<CGALPoint3, 4> pts0;
       std::array<CGALPoint3, 3> pts1;
       std::transform(
@@ -464,7 +464,7 @@ namespace CGALWrappers
     const std::array<Point<3>, 2> &vertices1,
     const double                   tol)
   {
-#  if DEAL_II_CGAL_VERSION_GTE(5, 1, 5)
+#  if DEAL_II_CGAL_VERSION_GTE(5, 5, 0)
     std::array<CGALPoint3_exact, 8> pts;
     std::transform(
       vertices0.begin(), vertices0.end(), pts.begin(), [&](const Point<3> &p) {
@@ -521,7 +521,7 @@ namespace CGALWrappers
     const std::array<Point<3>, 4> &vertices1,
     const double                   tol)
   {
-#  if DEAL_II_CGAL_VERSION_GTE(5, 1, 5)
+#  if DEAL_II_CGAL_VERSION_GTE(5, 5, 0)
     std::array<CGALPoint3_exact, 8> pts_hex;
     std::array<CGALPoint3_exact, 4> pts_quad;
     std::transform(

